examples/core/source.py

```diff
@@ -14,132 +17,135 @@
 # -
 
-# ## Create connection with speos rpc server
+# ### Define constants
+# Constants help ensure consistency and avoid repetition throughout the example.
 
-# +
-speos = Speos(host="localhost", port=50098)
-# -
+HOSTNAME = "localhost"
+GRPC_PORT = 50098  # Be sure the Speos GRPC Server has been started on this port.
+USE_DOCKER = True  # Set to False if you're running this example locally as a Notebook.
+IES = "IES_C_DETECTOR.ies"
 
-# ## Create a new project
+# ## Model Setup
+#
+# ### Load assets
+# The assets used to run this example are available in the
+# [PySpeos repository](https://github.com/ansys/pyspeos/) on GitHub.
+#
+# > **Note:** Make sure you
+# > have downloaded simulation assets and set ``assets_data_path``
+# > to point to the assets folder.
 
-# The only way to create a source, is to create it from a project.
+if USE_DOCKER:  # Running on the remote server.
+    assets_data_path = Path("/app") / "assets"
+else:
+    assets_data_path = Path("/path/to/your/download/assets/directory")
+
+# ### Connect to the RPC Server
+# This Python client connects to a server where the Speos engine
+# is running as a service. In this example, the server and
+# client are the same
+# machine.
+
+speos = Speos(host=HOSTNAME, port=GRPC_PORT)
+
+# ### Create a new project
+#
+# The only way to create a source using the core layer, is to create it from a project.
+# The ``Project`` class is instantiated by passing a ``Speos`` instance
 
-# +
 p = Project(speos=speos)
 print(p)
-# -
-
-# ## Create
 
-# Create locally
 
-# The mention "local: " is added when printing the source.
+# ### Source Creation
+#
+# #### Create locally
+# The mention "local: " is added when printing the source data and information is not yet
+# pushed to the RPC server
 
-# +
-intensity_file_path = str(assets_data_path / "IES_C_DETECTOR.ies")
 
+intensity_file_path = str(assets_data_path / IES)
 source1 = p.create_source(name="Luminaire.1", feature_type=SourceLuminaire)  # type luminaire
 source1.set_intensity_file_uri(uri=intensity_file_path)
 print(source1)
 # -
 
-# ## Push it to the server.
-
-# Now that it is committed to the server, the mention "local: " is no more present when printing the
+# #### Push it to the server.
+#
+# After it is committed to the server, the mention "local: " is no more present when printing the
 # source.
 
-# +
 source1.commit()
 print(source1)
-# -
-
-# ## Another example
 
+# #### Changing additional Source Properties
+#
 # Setting several more characteristics.
 
-# +
-intensity_file_path = str(assets_data_path / "IES_C_DETECTOR.ies")
 
+intensity_file_path = str(assets_data_path / IES)
 source2 = p.create_source(name="Luminaire.2", feature_type=SourceLuminaire)
 source2.set_intensity_file_uri(uri=intensity_file_path)
 source2.set_flux_radiant()  # select flux radiant with default value
 # choose the source location [Origin, Xvector, Yvector, Zvector]
 source2.set_axis_system(axis_system=[20, 50, 10, 1, 0, 0, 0, 1, 0, 0, 0, 1])
```

Comment:
to align with the changed done in combine-speos.py: https://github.com/ansys/pyspeos/blob/a8036a623c323b1aaa990f74ccd3de5c43fe2132/examples/workflow/combine-speos.py#L34, the coordinate system needs to be defined at the "Define constants" section

Reply:
not all are define there